pkg/kube/client.go

```diff
@@ -596,13 +615,32 @@ func (c *Client) waitForResources(timeout time.Duration, created Result) error {
 				if err != nil {
 					return false, err
 				}
+
 				for _, r := range rs.Items {
-					list, err := getPods(client, value.Namespace, r.Spec.Selector.MatchLabels)
+					currentReplicaSet, err := client.ReplicaSets(value.Namespace).Get(r.Name)
 					if err != nil {
 						return false, err
 					}
-					pods = append(pods, list...)
+					replicaSets = append(replicaSets, currentReplicaSet)
+				}
+
+				deployment, err := client.Deployments(value.Namespace).Get(value.Name)
+				if err != nil {
+					return false, err
+				}
+				// Find RS associated with deployment
+				newReplicaSet, err := deploymentutil.FindNewReplicaSet(deployment, replicaSets)
+				if err != nil {
+					return false, err
+				}
+				newDeployment := struct {
```

Comment:
Because you use this struct in 3 places, it might just be cleaner/easier to create it as a type